src/puya/ir/builder.py

```diff
@@ -2304,12 +2320,757 @@ def _iterate_tuple(
         self._seal(next_block)
         self.block_builder.activate_block(next_block)
 
-    def visit_array_append(self, expr: awst_nodes.ArrayAppend) -> TExpression:
-        raise TodoError(expr.source_location, "TODO: visit_new_struct")
-
     def visit_new_struct(self, expr: awst_nodes.NewStruct) -> TExpression:
         raise TodoError(expr.source_location, "TODO: visit_new_struct")
 
+    def _invoke_puya_lib_subroutine(
+        self,
+        *,
+        method_name: str,
+        module_name: str,
+        args: list[Value],
+        source_location: SourceLocation,
+    ) -> InvokeSubroutine:
+        sub = next(
```

Comment:
With a small refactor can use the context.resolve_function_reference to resolve these instead. I will push a commit for this